File: app/src/main/java/com/fed/notes/database/AppDatabase.java
package com.fed.notes.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.os.Environment;

import com.fed.notes.App;

import java.io.File;

/**
 * Created by Fedor SURIN on 26.10.2017.
 */

@Database(entities = Note.class, version = 1)
@TypeConverters(Converter.class)
public abstract class AppDatabase extends RoomDatabase {

    public abstract NoteDAO getNoteDao();
}

File: app/src/main/java/com/fed/notes/database/AppDatabase.kt
package com.fed.notes.database

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.TypeConverters
import android.arch.persistence.room.migration.Migration


@Database(entities = arrayOf(Note::class), version = 2)
@TypeConverters(Converter::class)
abstract class AppDatabase : RoomDatabase() {
    abstract val noteDao: NoteDAO

    companion object {
        @JvmField
        val MIGRATION_1_2: Migration = Migration1To2()
    }
}
